core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -473,7 +473,7 @@ default Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session,
                             .map(columnNamesByHandle::get)
                             .collect(toUnmodifiableList());
 
-                    return new ConnectorTableLayout(partitioning.getPartitioningHandle(), partitionColumns);
+                    return new ConnectorTableLayout(partitioning.getPartitioningHandle(), partitionColumns, false);
```

Comment:
leave the old methods and leverage them. avoid unnecessary changes.

we will deprecate these contrctors later, and update all the code places
keeping the changes separate helps reviewers focus on the meaty parts.